src/lava/lib/optimization/solvers/generic/solver.py

```diff
@@ -99,6 +110,19 @@ def run_cfg(self):
     def run_cfg(self, value):
         self._run_cfg = value
 
+    @property
+    def hyperparameters(self):
+        return self._hyperparameters
+
+    @hyperparameters.setter
+    def hyperparameters(self,
+                        value: ty.Dict[str, ty.Union[int, npt.ArrayLike]]):
+        self._hyperparameters = value
+
+    @property
+    def last_run_report(self):
+        return self._report
+

```

Comment:
Please add a tune method to the solver which uses the SolverTunner to get solving hyperparameters and set the hyperparameters property of OptimizationSolver (self).